### ISSUE TITLE
compile issues with nightly build

### ISSUE THREAD

**Issue by chilina** [openframeworks/openFrameworks#8102]
From forum thread here:  https://forum.openframeworks.cc/t/tesselator-h-and-compiling-the-rpi-nightly-build/43856

With of_v20240901_linuxarmv6l_release, compiling with the compileOF.sh script fails with:
`../../../libs/openFrameworks/utils/ofConstants.h:306:10: fatal error: tesselator.h: No such file or directory
  306 | #include <tesselator.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.`

Trying to compile on an RPi3B with 32-bit Bullseye, following the setup guide (which is very nicely written now and I'm finding it very easy to follow).

@dimitre thinks that maybe the libs path isn't getting included in the makefile.

**Comment by paolo-scoppola** [comment]
Same problem with of_v20240902_linuxaarch64_release.tar.gz 

Testing on Ubuntu 22.04 on virtual machine UTM on Mac M1 Pro

I run the download_libs.sh script then install_dependencies.sh and install_codecs.sh. Finally I run compileOF.sh and I get

find: '../../../libs/*/include': No such file or directory
find: '../../../libs/*/include': No such file or directory
find: '../../../libs/*/include': No such file or directory

....

../../../libs/openFrameworks/utils/ofConstants.h:306:10: fatal error: tesselator.h: No such file or directory

**Comment by dimitre** [comment]
Github actions seems to be passing (green) even when failing, 
we can see this in any recent rpi-build (armv6l, armhf, arm-linux-gnueabihf, 1)
```
../../../libs/openFrameworksCompiled/lib/linuxarmv6l/obj/Debug/libs/openFrameworks/math/ofMatrix4x4.o -c ../../../libs/openFrameworks/math/ofMatrix4x4.cpp
  In file included from ../../../libs/openFrameworks/math/ofMatrix4x4.h:13,
                   from ../../../libs/openFrameworks/math/ofMatrix4x4.cpp:1:
  ../../../libs/openFrameworks/math/ofVec4f.h:5:10: fatal error: glm/vec4.hpp: No such file or directory
      5 | #include <glm/vec4.hpp>
        |          ^~~~~~~~~~~~~~
  compilation terminated.
  make[2]: *** [makefileCommon/compile.core.mk:259: ../../../libs/openFrameworksCompiled/lib/linuxarmv6l/obj/Debug/libs/openFrameworks/math/ofMatrix4x4.o] Error 1
  make[2]: *** Waiting for unfinished jobs....
  In file included from ../../../libs/openFrameworks/math/ofVec2f.cpp:1:
  ../../../libs/openFrameworks/math/ofVec2f.h:8:10: fatal error: glm/vec2.hpp: No such file or directory
      8 | #include <glm/vec2.hpp>
        |          ^~~~~~~~~~~~~~
  compilation terminated.
  make[2]: *** [makefileCommon/compile.core.mk:259: ../../../libs/openFrameworksCompiled/lib/linuxarmv6l/obj/Debug/libs/openFrameworks/math/ofVec2f.o] Error 1
  make[1]: *** [makefileCommon/compile.core.mk:233: Debug] Error 2
  make[1]: Leaving directory '/home/runner/work/openFrameworks/openFrameworks/libs/openFrameworksCompiled/project'
  make: *** [../../../libs/openFrameworksCompiled/project/makefileCommon/compile.project.mk:141: Debug] Error 2
**** Building allAddonsExample ****
```

**Comment by paolo-scoppola** [comment]
Thanks @dimitre but in my case the compilation stops and no binary file is created.

```
paolo@paolo-QEMU-Virtual-Machine:~/Scrivania/of_v20240902_linuxaarch64_release/scripts/linux$ sudo ./buildAllExamples.sh
category ../../examples/computer_vision
-----------------------------------------------------------------
building ../../examples/computer_vision/opencvExample
make: Entering directory '/home/paolo/Scrivania/of_v20240902_linuxaarch64_release/examples/computer_vision/opencvExample'
Compiling OF library for Debug
make -j2 -C ../../../libs/openFrameworksCompiled/project/ Debug PLATFORM_OS=Linux ABIS_TO_COMPILE_DEBUG=""
make[1]: Entering directory '/home/paolo/Scrivania/of_v20240902_linuxaarch64_release/libs/openFrameworksCompiled/project'
\U0001f4be CXX = g++
\U0001f4be CC = cc
\U0001f4be CXX = g++
\U0001f4be CC = cc
find: '../../../libs/*/include': No such file or directory
find: '../../../libs/*/include': No such file or directory
find: '../../../libs/*/include': No such file or directory
find: '../../../libs/*/include': No such file or directory
Compiling ../../../libs/openFrameworks/graphics/ofPixels.cpp
find: '../../../libs/*/include': No such file or directory
g++ -g3 -DDEBUG -Wall -Werror=return-type -DGCC_HAS_REGEX -std=c++20 -march=armv8-a -mcpu=cortex-a72 -mtune=cortex-a72 -fPIC -ftree-vectorize -Wno-psabi -pipe -DOF_USING_GTK -DOF_USING_GTK -DOF_USING_MPG123 -pthread -D__LINUX_ALSA__ -D__LINUX_PULSE__ -D__UNIX_JACK__ -D_REENTRANT -pthread -I/usr/include/gstreamer-1.0 -I/usr/include/orc-0.4 -I/usr/include/gstreamer-1.0 -I/usr/include/aarch64-linux-gnu -I/usr/include/opus -I/usr/include/AL -I/usr/include/aarch64-linux-gnu -I/usr/include/rtaudio -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/aarch64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/aarch64-linux-gnu -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I../../../libs/openFrameworks -I../../../libs/openFrameworks/graphics -I../../../libs/openFrameworks/app -I../../../libs/openFrameworks/utils -I../../../libs/openFrameworks/sound -I../../../libs/openFrameworks/events -I../../../libs/openFrameworks/types -I../../../libs/openFrameworks/gl -I../../../libs/openFrameworks/math -I../../../libs/openFrameworks/communication -I../../../libs/openFrameworks/3d -I../../../libs/openFrameworks/video -Wall -Werror=return-type -DGCC_HAS_REGEX -std=c++20 -MMD -MP -MF ../../../libs/openFrameworksCompiled/lib/linuxaarch64/obj/Debug/libs/openFrameworks/graphics/ofPixels.d -MT../../../libs/openFrameworksCompiled/lib/linuxaarch64/obj/Debug/libs/openFrameworks/graphics/ofPixels.o -o ../../../libs/openFrameworksCompiled/lib/linuxaarch64/obj/Debug/libs/openFrameworks/graphics/ofPixels.o -c ../../../libs/openFrameworks/graphics/ofPixels.cpp
Compiling ../../../libs/openFrameworks/graphics/ofPath.cpp
find: '../../../libs/*/include': No such file or directory
g++ -g3 -DDEBUG -Wall -Werror=return-type -DGCC_HAS_REGEX -std=c++20 -march=armv8-a -mcpu=cortex-a72 -mtune=cortex-a72 -fPIC -ftree-vectorize -Wno-psabi -pipe -DOF_USING_GTK -DOF_USING_GTK -DOF_USING_MPG123 -pthread -D__LINUX_ALSA__ -D__LINUX_PULSE__ -D__UNIX_JACK__ -D_REENTRANT -pthread -I/usr/include/gstreamer-1.0 -I/usr/include/orc-0.4 -I/usr/include/gstreamer-1.0 -I/usr/include/aarch64-linux-gnu -I/usr/include/opus -I/usr/include/AL -I/usr/include/aarch64-linux-gnu -I/usr/include/rtaudio -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/aarch64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/aarch64-linux-gnu -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I../../../libs/openFrameworks -I../../../libs/openFrameworks/graphics -I../../../libs/openFrameworks/app -I../../../libs/openFrameworks/utils -I../../../libs/openFrameworks/sound -I../../../libs/openFrameworks/events -I../../../libs/openFrameworks/types -I../../../libs/openFrameworks/gl -I../../../libs/openFrameworks/math -I../../../libs/openFrameworks/communication -I../../../libs/openFrameworks/3d -I../../../libs/openFrameworks/video -Wall -Werror=return-type -DGCC_HAS_REGEX -std=c++20 -MMD -MP -MF ../../../libs/openFrameworksCompiled/lib/linuxaarch64/obj/Debug/libs/openFrameworks/graphics/ofPath.d -MT../../../libs/openFrameworksCompiled/lib/linuxaarch64/obj/Debug/libs/openFrameworks/graphics/ofPath.o -o ../../../libs/openFrameworksCompiled/lib/linuxaarch64/obj/Debug/libs/openFrameworks/graphics/ofPath.o -c ../../../libs/openFrameworks/graphics/ofPath.cpp
In file included from ../../../libs/openFrameworks/graphics/ofPath.h:3,
                 from ../../../libs/openFrameworks/graphics/ofPath.cpp:1:
../../../libs/openFrameworks/graphics/ofPolyline.h:8:10: fatal error: glm/fwd.hpp: No such file or directory
    8 | #include <glm/fwd.hpp>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [makefileCommon/compile.core.mk:259: ../../../libs/openFrameworksCompiled/lib/linuxaarch64/obj/Debug/libs/openFrameworks/graphics/ofPath.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from ../../../libs/openFrameworks/math/ofMathConstants.h:4,
                 from ../../../libs/openFrameworks/graphics/ofGraphicsConstants.h:3,
                 from ../../../libs/openFrameworks/graphics/ofPixels.cpp:1:
../../../libs/openFrameworks/utils/ofConstants.h:306:10: fatal error: tesselator.h: No such file or directory
  306 | #include <tesselator.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [makefileCommon/compile.core.mk:259: ../../../libs/openFrameworksCompiled/lib/linuxaarch64/obj/Debug/libs/openFrameworks/graphics/ofPixels.o] Error 1
make[1]: *** [makefileCommon/compile.core.mk:233: Debug] Error 2
make[1]: Leaving directory '/home/paolo/Scrivania/of_v20240902_linuxaarch64_release/libs/openFrameworksCompiled/project'
make: *** [../../../libs/openFrameworksCompiled/project/makefileCommon/compile.project.mk:141: Debug] Error 2
make: Leaving directory '/home/paolo/Scrivania/of_v20240902_linuxaarch64_release/examples/computer_vision/opencvExample'
error compiling ../../examples/computer_vision/opencvExample

```

**Comment by thebigG** [comment]
Ran into a similar issue:
```
In file included from ../../../libs/openFrameworks/sound/ofSoundBaseTypes.h:4,
                 from ../../../libs/openFrameworks/sound/ofSoundPlayer.h:3,
                 from ../../../libs/openFrameworks/sound/ofSoundPlayer.cpp:1:
../../../libs/openFrameworks/utils/ofConstants.h:306:10: fatal error: tesselator.h: No such file or directory
  306 | #include <tesselator.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [makefileCommon/compile.core.mk:259: ../../../libs/openFrameworksCompiled/lib/linux64/obj/Debug/libs/openFrameworks/sound/ofSoundPlayer.o] Error 1
make[1]: *** Waiting for unfinished jobs....
In file included from ../../../libs/openFrameworks/sound/ofFmodSoundPlayer.h:4,
                 from ../../../libs/openFrameworks/sound/ofFmodSoundPlayer.cpp:1:
../../../libs/openFrameworks/utils/ofConstants.h:306:10: fatal error: tesselator.h: No such file or directory
  306 | #include <tesselator.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [makefileCommon/compile.core.mk:259: ../../../libs/openFrameworksCompiled/lib/linux64/obj/Debug/libs/openFrameworks/sound/ofFmodSoundPlayer.o] Error 1
In file included from ../../../libs/openFrameworks/utils/ofFileUtils.h:4,
                 from ../../../libs/openFrameworks/utils/ofLog.h:4,
                 from ../../../libs/openFrameworks/sound/ofSoundBuffer.cpp:10:
../../../libs/openFrameworks/utils/ofConstants.h:306:10: fatal error: tesselator.h: No such file or directory
  306 | #include <tesselator.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [makefileCommon/compile.core.mk:259: ../../../libs/openFrameworksCompiled/lib/linux64/obj/Debug/libs/openFrameworks/sound/ofSoundBuffer.o] Error 1
In file included from ../../../libs/openFrameworks/sound/ofRtAudioSoundStream.h:2,
                 from ../../../libs/openFrameworks/sound/ofRtAudioSoundStream.cpp:1:
../../../libs/openFrameworks/utils/ofConstants.h:306:10: fatal error: tesselator.h: No such file or directory
  306 | #include <tesselator.h>
      |          ^~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [makefileCommon/compile.core.mk:259: ../../../libs/openFrameworksCompiled/lib/linux64/obj/Debug/libs/openFrameworks/sound/ofRtAudioSoundStream.o] Error 1
make: *** [makefileCommon/compile.core.mk:233: Debug] Error 2
there has been a problem compiling Debug OF library

```

System info:

```
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os

```

**Comment by dimitre** [comment]
Hello @danoli3 I wonder if it is apothecary related, or maybe installation scripts
@thebigG did you run the installation scripts? can you please list the files in the folder openFrameworks/libs/tess2 ?

**Comment by paolo-scoppola** [comment]
Hi,

I **fixed** the issue with tessellator.h - it's due to the wrong position of libs after the download. They should be in OF/libs but I've found them in OF/libs/openFrameworksLibs_bleeding_linuxaarch64. So if you move glm, json,..., tess2 and the others to OF/libs it compiles OF.

Now I've issue with compiling examples. it complains about **lblas** and **llapack**

```
./buildAllExamples.sh 
category ../../examples/computer_vision
-----------------------------------------------------------------
building ../../examples/computer_vision/opencvExample
make: Entering directory '/home/paolo/Scrivania/of_v20240902_linuxaarch64_release/examples/computer_vision/opencvExample'
Compiling OF library for Debug
make -j2 -C ../../../libs/openFrameworksCompiled/project/ Debug PLATFORM_OS=Linux ABIS_TO_COMPILE_DEBUG=""
make[1]: Entering directory '/home/paolo/Scrivania/of_v20240902_linuxaarch64_release/libs/openFrameworksCompiled/project'
\U0001f4be CXX = g++
\U0001f4be CC = cc
\U0001f4be CXX = g++
\U0001f4be CC = cc
\U0001f4be CXX = g++
\U0001f4be CC = cc
Done!
make[1]: Leaving directory '/home/paolo/Scrivania/of_v20240902_linuxaarch64_release/libs/openFrameworksCompiled/project'


Compiling opencvExample for Debug
make[1]: Entering directory '/home/paolo/Scrivania/of_v20240902_linuxaarch64_release/examples/computer_vision/opencvExample'
ADDON_INCLUDES_EXCLUDE: libs/opencv libs/opencv/%
ADDON_SOURCES_EXCLUDE: 
\U0001f517 Linking bin/opencvExample_debug for linuxaarch64
g++ -o bin/opencvExample_debug  obj/linuxaarch64/Debug/src/main.o obj/linuxaarch64/Debug/src/ofApp.o  ../../../addons/obj/linuxaarch64/Debug/ofxOpenCv/src/ofxCvGrayscaleImage.o ../../../addons/obj/linuxaarch64/Debug/ofxOpenCv/src/ofxCvHaarFinder.o ../../../addons/obj/linuxaarch64/Debug/ofxOpenCv/src/ofxCvImage.o ../../../addons/obj/linuxaarch64/Debug/ofxOpenCv/src/ofxCvContourFinder.o ../../../addons/obj/linuxaarch64/Debug/ofxOpenCv/src/ofxCvFloatImage.o ../../../addons/obj/linuxaarch64/Debug/ofxOpenCv/src/ofxCvShortImage.o ../../../addons/obj/linuxaarch64/Debug/ofxOpenCv/src/ofxCvColorImage.o  ../../../libs/openFrameworksCompiled/lib/linuxaarch64/libopenFrameworksDebug.a  -Wl,-rpath=./libs:./bin/libs:./ -Wl,--as-needed -Wl,--gc-sections -no-pie -lblas -llapack ../../../libs/kiss/lib/linuxaarch64/libkiss.a ../../../libs/libxml2/lib/linuxaarch64/libxml2.a ../../../libs/svgtiny/lib/linuxaarch64/libsvgtiny.a ../../../libs/tess2/lib/linuxaarch64/libtess2.a -lz -lgstapp-1.0 -lgstvideo-1.0 -lgstbase-1.0 -lgstreamer-1.0 -ludev -lfontconfig -lfreetype -lsndfile -lopenal -lssl -lcrypto -lcurl -lglfw -lrtaudio -lpulse-simple -pthread -lpulse -pthread -lasound -lGL -lGLEW -lGL -lX11 -lGLU -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lmpg123 -lGLESv1_CM -lGLESv2 -lEGL -lopencv_stitching -lopencv_alphamat -lopencv_aruco -lopencv_barcode -lopencv_bgsegm -lopencv_bioinspired -lopencv_ccalib -lopencv_dnn_objdetect -lopencv_dnn_superres -lopencv_dpm -lopencv_face -lopencv_freetype -lopencv_fuzzy -lopencv_hdf -lopencv_hfs -lopencv_img_hash -lopencv_intensity_transform -lopencv_line_descriptor -lopencv_mcc -lopencv_quality -lopencv_rapid -lopencv_reg -lopencv_rgbd -lopencv_saliency -lopencv_shape -lopencv_stereo -lopencv_structured_light -lopencv_phase_unwrapping -lopencv_superres -lopencv_optflow -lopencv_surface_matching -lopencv_tracking -lopencv_highgui -lopencv_datasets -lopencv_text -lopencv_plot -lopencv_ml -lopencv_videostab -lopencv_videoio -lopencv_viz -lopencv_wechat_qrcode -lopencv_ximgproc -lopencv_video -lopencv_xobjdetect -lopencv_objdetect -lopencv_calib3d -lopencv_imgcodecs -lopencv_features2d -lopencv_dnn -lopencv_flann -lopencv_xphoto -lopencv_photo -lopencv_imgproc -lopencv_core -lglut -lX11 -lXrandr -lXxf86vm -lXi -lXcursor -ldl -lpthread -lfreeimage -lpugixml -luriparser -lXinerama
/usr/bin/ld: cannot find -lblas: No such file or directory
/usr/bin/ld: cannot find -llapack: No such file or directory
collect2: error: ld returned 1 exit status
make[1]: *** [../../../libs/openFrameworksCompiled/project/makefileCommon/compile.project.mk:411: bin/opencvExample_debug] Error 1
make[1]: Leaving directory '/home/paolo/Scrivania/of_v20240902_linuxaarch64_release/examples/computer_vision/opencvExample'
make: *** [../../../libs/openFrameworksCompiled/project/makefileCommon/compile.project.mk:145: Debug] Error 2
make: Leaving directory '/home/paolo/Scrivania/of_v20240902_linuxaarch64_release/examples/computer_vision/opencvExample'
error compiling ../../examples/computer_vision/opencvExample
```

**Comment by paolo-scoppola** [comment]
I tried the other script 

`./buildAllRPIExamples.sh`

it compiles some examples and for others it complains about an other path 

`../../../libs/*/lib/linuxaarch64/*.so`

I think that in the of_v20240902_linuxaarch64_release  night build or maybe even in others there are some issues with the folders structure, i checked other older night builds and I found differences.

**Comment by chilina** [comment]
Hey way to go @paolo-scoppola !  You know I tried moving all of the libs from the "bleeding edge" folder directly into /libs for the armv6 nightly , and I got a different set of compile errors.  Also, it looks like you're trying to compile arch linux?  And since maybe its not just an armv6 issue, I think I'll change the title back to something more generic.

**Comment by thebigG** [comment]
Turns out I hadn't run one of he scripts called "./download_libs.sh". After I ran that, I was able to build :+1: 

For anybody having issues this is what I did:

```
git submodule update --init --recursive
cd openFrameworks/scripts/linux
./compileOF.sh -j4
./download_libs.sh
cd ubuntu
./compileOF.sh -j4
cd scripts/linux
./compilePG.sh
projectGenerator -o  **apps/myApps/newExample**
```

**Comment by thebigG** [comment]
NOTE:Remember to add "export PG_OF_PATH=/home/lgomez/shaders/openFrameworks" to bashrc/zshrc.

**Comment by etlaM21** [comment]
Ran into another issue, after fixing the libs issue by moving the folders to the right directory as pointed out by @paolo-scoppola 

I ran `download_libs.sh`, `debian/install_dependencies.sh` and `debian/install_codecs.sh`.

Trying to compile Nightly on my Raspberry PI B+ V1.2

When I run `sudo ./compileOF.sh`, it aborts with:

`../../../libs/openFrameworks/sound/ofSoundStream.cpp: In constructor 'ofSoundStream::ofSoundStream()':
../../../libs/openFrameworks/sound/ofSoundStream.cpp:26:30: error: 'ofRtAudioSoundStream' was not declared in this scope; did you mean 'ofBaseSoundStream'?
   26 | #define OF_SOUND_STREAM_TYPE ofRtAudioSoundStream
      |                              ^~~~~~~~~~~~~~~~~~~~
../../../libs/openFrameworks/sound/ofSoundStream.cpp:181:49: note: in expansion of macro 'OF_SOUND_STREAM_TYPE'
  181 |                 setSoundStream(std::make_shared<OF_SOUND_STREAM_TYPE>());
      |                                                 ^~~~~~~~~~~~~~~~~~~~
../../../libs/openFrameworks/sound/ofSoundStream.cpp:181:70: error: no matching function for call to 'make_shared<<expression error> >()'
  181 |                 setSoundStream(std::make_shared<OF_SOUND_STREAM_TYPE>());
      |                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from /usr/include/c++/12/memory:77,
                 from ../../../libs/openFrameworks/utils/ofConstants.h:17,
                 from ../../../libs/openFrameworks/sound/ofSoundBaseTypes.h:4,
                 from ../../../libs/openFrameworks/app/ofBaseApp.h:3,
                 from ../../../libs/openFrameworks/sound/ofSoundStream.h:3,
                 from ../../../libs/openFrameworks/sound/ofSoundStream.cpp:1:
/usr/include/c++/12/bits/shared_ptr.h:1005:5: note: candidate: 'template<class _Tp, class ... _Args> std::shared_ptr<typename std::enable_if<(! std::is_array< <template-parameter-1-1> >::value), _Tp>::type> std::make_shared(_Args&& ...)'
 1005 |     make_shared(_Args&&... __args)
      |     ^~~~~~~~~~~
/usr/include/c++/12/bits/shared_ptr.h:1005:5: note:   template argument deduction/substitution failed:
../../../libs/openFrameworks/sound/ofSoundStream.cpp:181:70: error: template argument 1 is invalid
  181 |                 setSoundStream(std::make_shared<OF_SOUND_STREAM_TYPE>());
      |                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
make[1]: *** [makefileCommon/compile.core.mk:261: ../../../libs/openFrameworksCompiled/lib/linuxarmv6l/obj/Debug/libs/openFrameworks/sound/ofSoundStream.o] Error 1
make: *** [makefileCommon/compile.core.mk:233: Debug] Error 2`

**Comment by dimitre** [comment]
@etlaM21 are you using nightly build? or 0.12.0 ?

**Comment by etlaM21** [comment]
> @etlaM21 are you using nightly build? or 0.12.0 ?

I'm trying to compile the nightly build. Should be the one from yesterday, 14.10.

**Comment by dimitre** [comment]
@etlaM21 please test some changes in ofRtAudioSoundStream.cpp
substitute the first  7 lines with this:
```c++
#include "ofRtAudioSoundStream.h"
#include "ofConstants.h"
#if defined(OF_RTAUDIO)
#include "ofUtils.h"
#include "ofAppRunner.h"
#include "ofLog.h"
#include <RtAudio.h>
```
cc: @danoli3

**Comment by etlaM21** [comment]
@dimitre I just tried these changes and still get the same error message.

**Comment by roymacdonald** [comment]
I think that the problem is that the target is not being properly setup by OF. There are a bunch of preprocessor directives that are setup depending on the platform, os, etc, and because of that the RTAudio is not being added to the raspi build.

**Comment by danoli3** [comment]
Let me look into this today, I think the best solution is to actually compile GLM headers as per spec rather than just copy from git repo as before. This way headers are placed / macros defined and copied to the spec for platform

**Comment by danoli3** [comment]
I think the whole install / process for RPi and Linux is a bit all over the shot at the moment though. Like really need to fix the whole install process. Confusing as f

**Comment by dimitre** [comment]
@danoli3 I've noticed RPI runners are now completing with success. 
are RPI libs fixed?

**Comment by ofTheo** [comment]
@chilina - can you try the latest nightly build and see if this is still an issue?

**Comment by chilina** [comment]
Hey I just tried the latest nightly and still have the original issue.  But maybe I'll do a clean install of Bullseye and try again.

**Comment by dimitre** [comment]
Hey @chilina I think this is finally fixed by https://github.com/openframeworks/openFrameworks/pull/8207/

**Comment by ofTheo** [comment]
closed by #8207
